### PR TITLE
fix(OH2-366): restrict userName characters

### PR DESCRIPTION
**Pull request by gasp**
OH2-366

restrict userName characters
username-test

### CODE REVIEW

File: src/components/accessories/admin/users/newUser/validation.test.ts [hunk 1]
```diff
@@ -17,3 +17,22 @@ describe("password rules", () => {
...
+    expect(userNameRules.test("john-doe")).toBeTruthy();
+    expect(userNameRules.test("john_doe")).toBeTruthy();
+  });
+  it("should filter out", () => {
+    expect(userNameRules.test("Johndoe")).toBeFalsy();
+    expect(userNameRules.test("johnDoe")).toBeFalsy();
+    expect(userNameRules.test("john doe")).toBeFalsy();
+    expect(userNameRules.test("john/doe")).toBeFalsy();
+    expect(userNameRules.test("すず")).toBeFalsy();
+    expect(userNameRules.test("j̵̨̨̧͖̠̩̤̗̟̲̯̭̫̰͆͛̏͛͒́̂̔̅͘͘̚̕͝͝ȯ̵̫̭̮̖̀̓̾̉͋͋̌̇͘h̶̡̢̡̜̻̥͙̳͉̰̟̬͚̍̃̽̎͒̋̄̔͋͘͝͝ͅn̷̜̠̰͍̤̰̺̠͌̌̒͑̓̌̂̒͗͒͗̐͝͝͠")).toBeFalsy();
```
Comment (SilverD3) [inline src/components/accessories/admin/users/newUser/validation.test.ts:36]: What is this ?

**Review by SilverD3 — COMMENTED** [review]
LGTM!

**Comment by SilverD3** [comment]
I recognize that this comment doesn't fall within the scope of this PR, but I just noticed this possible design improvement.
User Forml
1. `password` and `confirmPassword` fields can be on the line (two columns grid).
2. `description` should be a textarea

File: src/components/accessories/admin/users/newUser/validation.test.ts (hunk 1 above)
Comment (gasp) [reply src/components/accessories/admin/users/newUser/validation.test.ts:36]: It's a common hacking technique using special characters to mess up with the filters and produces visually appealing results. It has also been used as a form of art by visual artists such as Rosa Menkman.

**Review by mwithi — APPROVED** [review]
Reviewed and approved by @SilverD3

**Pull request merged**